config/monitoring/metrics/opencensus/100-ocagent.yaml

```diff
@@ -0,0 +1,104 @@
+# Copyright 2018 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Based on https://github.com/census-instrumentation/opencensus-service/raw/master/example/k8s.yaml
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: opencensus-agent-config
+  namespace: knative-monitoring
+  labels:
+    serving.knative.dev/release: devel
+data:
+  opencensus-agent-config: |
+    receivers:
+      opencensus:
+        address: ":55678"
+    #   jaeger: {}
+    #   zipkin: {}
+    exporters: {}
+    # stackdriver:
+    #   project: "my-gcp-project"  # Optional, defaults to node project
+    #   metric_prefix: "your-metric-prefix"
+    #   enable_metrics: true
+    #   enable_tracing: true
+    # prometheus:
+    #   namespace: "namespace"
+    #   address: ":8888"
+---
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: opencensus-agent
+  namespace: knative-monitoring
+  labels:
+    serving.knative.dev/release: devel
+spec:
+  template:
+    metadata:
+      labels:
+        component: opencensus-agent
+        serving.knative.dev/release: devel
+    spec:
+      containers:
+        - command:
+            - "/ocagent_linux"
+            - "--config=/conf/opencensus-agent-config.yaml"
+          image: omnition/opencensus-agent:0.1.6
+          name: opencensus-agent
+          resources:
+            limits:
+              cpu: 500m
+              memory: 500Mi
+            requests:
+              cpu: 100m
+              memory: 100Mi
+          ports:
+            - containerPort: 55678
+              name: collect
+            - containerPort: 55679
+              name: zpages
+            # For Prometheus scrapes
+            - containerPort: 8888
```

Comment:
The ports are in the ConfigMap, IIRC. The config is read by the `ocagent` process once at startup, from what I can tell, so changing the values in the opencensus ConfigMap requires doing a rolling restart of the DaemonSet in this sample configuration.

I exposed the port here at all times because I figured that the danger of exposing a port that the process wasn't listening to was low, and the benefit in terms of being able to set up Prometheus scraping was high.